src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,131 @@
-import { useState } from 'react';
+import React, { useEffect, useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd(movie: Movie): void;
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+  const [newMovie, setNewMovie] = useState({
+    title: '',
+    description: '',
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+  });
+  const [buttonDisabled, setButtonDisabled] = useState(true);
+  const [isImdbUrlWrong, setIsImdbUrlWrong] = useState(false);
+  const [isImgUrlWrong, setIsImgUrlWrong] = useState(false);
+
+  const reset = () => {
+    setNewMovie({
+      title: '',
+      description: '',
+      imgUrl: '',
+      imdbUrl: '',
+      imdbId: '',
+    });
+  };
+
+  const onSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    // eslint-disable-next-line max-len
+    const pattern = /^((([A-Za-z]{3,9}:(?:\/\/)?)(?:[-;:&=+$,\w]+@)?[A-Za-z0-9.-]+|(?:www\.|[-;:&=+$,\w]+@)[A-Za-z0-9.-]+)((?:\/[+~%/.\w-_]*)?\??(?:[-+=&;%@,.\w_]*)#?(?:[,.!/\\\w]*))?)$/;
+
+    let isUrlWrong = false;
+
+    if (!pattern.test(newMovie.imdbUrl.trim())) {
+      setIsImdbUrlWrong(true);
+      isUrlWrong = true;
+    }
+
+    if (!pattern.test(newMovie.imgUrl.trim())) {
+      setIsImgUrlWrong(true);
+      isUrlWrong = true;
+    }
+
+    if (isUrlWrong) {
+      return;
+    }
+
+    setCount(count + 1);
+
+    onAdd(newMovie);
+
+    reset();
+  };
+
+  useEffect(() => {
+    if (newMovie.title.trim() && newMovie.imdbId.trim()
+    && newMovie.imdbUrl.trim() && newMovie.imgUrl.trim()) {
+      setButtonDisabled(false);
+    } else {
+      setButtonDisabled(true);
+    }
+  });
```

Comment:
Redundant useEffect and useState 
```suggestion
const isButtonDisabled = newMovie.title.trim() 
  && newMovie.imdbId.trim() 
  && newMovie.imdbUrl.trim() 
  && newMovie.imgUrl.trim()
```